Check return value for getProduct() in getPrice().

pbaylies opened the pull request:
If getProduct() returns a NULL value then this code will generate a PHP warning and potentially crash the page; let's not do that, and check our return values before we try to call methods on them.

pbaylies committed:
Check return value for getProduct() in getPrice().

If getProduct() returns a NULL value then this code will generate a PHP warning and potentially crash the page; let's not do that, and check our return values before we try to call methods on them.

antonkril reviewed (changes requested):
Please consider saving results of getCustomOption() and getProduct() calls. Also please address static test feedback.

pbaylies committed:
Update Price.php

Save results of methods called and clean up code in response to feedback and tests.

antonkril commented on app/code/Magento/ConfigurableProduct/Model/Product/Type/Configurable/Price.php:
```diff
@@ -39,10 +39,15 @@ public function getFinalPrice($qty, $product)
-            return $product->getCustomOption('simple_product')->getProduct()->getPrice();
-        } else {
-            return 0;
+        if ( !empty( $product ) ) {
```
This check is not required

vrann commented:
@pbaylies Thank you for contribution. Can you please fix following code styles issues:

```
FILE: ...e/Magento/ConfigurableProduct/Model/Product/Type/Configurable/Price.php
--------------------------------------------------------------------------------
FOUND 6 ERROR(S) AFFECTING 3 LINE(S)
--------------------------------------------------------------------------------
 42 | ERROR | Expected 0 spaces after opening bracket; 1 found
 42 | ERROR | Expected 0 spaces before closing bracket; 1 found
 44 | ERROR | Expected 0 spaces after opening bracket; 1 found
 44 | ERROR | Expected 0 spaces before closing bracket; 1 found
 46 | ERROR | Expected 0 spaces after opening bracket; 1 found
 46 | ERROR | Expected 0 spaces before closing bracket; 1 found
--------------------------------------------------------------------------------
```

pbaylies commented:
@vrann I can't reproduce this; I don't see any extra spaces at the end of those lines on my committed file.

vrann commented:
@pbaylies 

these lines
```
if ( !empty( $product ) ) {
if ( !empty( $simpleProductOption ) ) {
if ( !empty( $simpleProduct ) ) {
```
should look like this:
```
if (!empty($product)) {
if (!empty($simpleProductOption)) {
if (!empty($simpleProduct)) {
```

pbaylies committed:
Update Price.php

Coding style updates.

pbaylies commented:
@vrann Done, thanks!

vrann commented:
@pbaylies Thank you!

vrann commented:
@pbaylies merged!
